### PR TITLE
Case Search & Claim feature tests

### PR DESCRIPTION
**Pull request by phillipm**
Tests case search and claim across different users. Checking that you can't claim your own case, empty results are handled properly, closing a claimed case also removes the case from the original creator's case list.

UI tests for https://github.com/dimagi/commcare-odk/pull/1059

### CODE REVIEW

File: features/step_definitions/navigation.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Then (/^I logout/) do
+  if current_activity() == "CommCareHomeActivity"
+    touch("android.support.v7.widget.CardView index:4")
```
Comment (wpride) [inline features/step_definitions/navigation.rb]: OOC what's your policy for using element index vs. contains text?

File: features/case_claim.feature [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    Then I login with username "test" and password "123"
+    Then I scroll until I see the "Start" text
+    Then I touch the "Start" text
+    Then I select module "Pregnancy"
+    Then I select module "Registration"
+    Then I wait for form entry
+    Then I enter text "luca"
+    Then Next
+    Then I enter text "boston"
+    Then Next
```
Comment (wpride) [inline features/case_claim.feature]: A "next [x] times" helper might be nice at some point

File: features/step_definitions/navigation.rb (hunk 1 above)
Comment (phillipm) [inline features/step_definitions/navigation.rb]: The 'contains text' approach seems easier to understand, but I've been running into strange issues so I wanted to try this out.

File: features/case_claim.feature (hunk 2 above)
Comment (phillipm) [inline features/case_claim.feature]: good call, implemented

**Comment by phillipm** [comment]
This is ready to go now.

**Comment by wpride** [comment]
@phillipm this is really, really great cool.

**Pull request merged**